### ISSUE TITLE
add_kubernetes_metadata processor panic when closing

### ISSUE THREAD

**Issue by AndersonQ** [elastic/beats#41417]
The add_kubernetes_metadata processor might panic when closing. Its `*kubernetesAnnotator.Close()` attempts to close its cache which might close twice the cache's done channel.

The panic 

For confirmed bugs, please report:
- Version: 8.6.2, likely on main as well
- Operating System: linux
- Discuss Forum URL: n/a
- Steps to Reproduce: unclear, but closing the add_kubernetes_metadata processor twice should cause it to panic

Panic stacktrace
```
harvester panic with: close of closed channel
goroutine 174 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x65
github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile.startHarvester.func1.1()
	github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile/harvester.go:167 +0x78
panic({0x55d415b71780, 0x55d416139050})
	runtime/panic.go:844 +0x258
github.com/elastic/beats/v7/libbeat/processors/add_kubernetes_metadata.(*cache).stop(...)
	github.com/elastic/beats/v7/libbeat/processors/add_kubernetes_metadata/cache.go:97
github.com/elastic/beats/v7/libbeat/processors/add_kubernetes_metadata.(*kubernetesAnnotator).Close(0xc00025bf00?)
	github.com/elastic/beats/v7/libbeat/processors/add_kubernetes_metadata/kubernetes.go:311 +0x4f
github.com/elastic/beats/v7/libbeat/processors.Close(...)
	github.com/elastic/beats/v7/libbeat/processors/processor.go:58
github.com/elastic/beats/v7/libbeat/publisher/processing.(*group).Close(0x5?)
	github.com/elastic/beats/v7/libbeat/publisher/processing/processors.go:96 +0x159
github.com/elastic/beats/v7/libbeat/processors.Close(...)
	github.com/elastic/beats/v7/libbeat/processors/processor.go:58
github.com/elastic/beats/v7/libbeat/publisher/processing.(*group).Close(0x0?)
	github.com/elastic/beats/v7/libbeat/publisher/processing/processors.go:96 +0x159
github.com/elastic/beats/v7/libbeat/processors.Close(...)
	github.com/elastic/beats/v7/libbeat/processors/processor.go:58
github.com/elastic/beats/v7/libbeat/publisher/pipeline.(*client).Close.func1()
	github.com/elastic/beats/v7/libbeat/publisher/pipeline/client.go:167 +0x2df
sync.(*Once).doSlow(0x0?, 0x0?)
	sync/once.go:68 +0xc2
sync.(*Once).Do(...)
	sync/once.go:59
github.com/elastic/beats/v7/libbeat/publisher/pipeline.(*client).Close(0x55d413b78466?)
	github.com/elastic/beats/v7/libbeat/publisher/pipeline/client.go:148 +0x59
github.com/elastic/beats/v7/filebeat/beater.(*countingClient).Close(0x55d413b783df?)
	github.com/elastic/beats/v7/filebeat/beater/channels.go:145 +0x22
github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile.startHarvester.func1({0x55d41617e508?, 0xc00085de00})
	github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile/harvester.go:219 +0x929
github.com/elastic/go-concert/unison.(*TaskGroup).Go.func1()
	github.com/elastic/go-concert@v0.2.0/unison/taskgroup.go:163 +0xc3
created by github.com/elastic/go-concert/unison.(*TaskGroup).Go
	github.com/elastic/go-concert@v0.2.0/unison/taskgroup.go:159 +0xca
```

**Comment by jlind23** [comment]
@AndersonQ is this really happening on main too?

**Comment by rdner** [comment]
I think it's a duplicate of https://github.com/elastic/beats/issues/34219
See a KB here as well https://support.elastic.dev/knowledge/view/127cc47b
The fix was released in 8.7
